.github/workflows/ci.yml

```
@@ -63,9 +63,31 @@ jobs:
       with:
         filters: .github/change-filters.yml
 
+
+  build-tket1-passes:
+    name: Build tket1-passes C++ library
+    runs-on: ubuntu-latest
```

This should be at least filtered by the rust changes flag
```suggestion
  build-tket1-passes:
    name: Build tket1-passes C++ library
    if: ${{ needs.changes.outputs.rust == 'true' }}
    runs-on: ubuntu-latest
```
Be sure to add the new dirs to `.github/change-filters.yml`.

My main issue is that it's adding 30% runtime to the critical path of _every_ PR required checks run.

Do you think caching will work well here?
Would it make sense to **only** build/test this on changes to `tket1-passes` (at least for the required checks)?

---

if we specify the version of tket to be building against, then we can just watch local files, cache the compiled library and only rebuild when one of our files locally is changed. I just don't know how/where to cache the library, but I'll look it up now.

---

ok, seems like caching is quite easy. Will implement that.